gpm_api/tests/test_utils/test_utils_checks.py

```diff
@@ -0,0 +1,741 @@
+        ds["time"] = time
+
+        return ds
+
+    @pytest.fixture
+    def ds_invalid(
+        self,
+        ds_valid: xr.Dataset,
+    ) -> xr.Dataset:
+        ds = ds_valid.copy(deep=True)
+
+        lon = ds["lon"].values.copy()
+        lon[0, self.invalid_idx] = np.nan
+        ds["lon"] = (("cross_track", "along_track"), lon)
+
+        return ds
+
+    def test_is_valid_geolocation(
+        self,
+        set_is_orbit_to_true: None,
+        ds_valid: xr.Dataset,
+        ds_invalid: xr.Dataset,
+    ) -> None:
+        """Test _is_valid_geolocation"""
+
+        # Valid
+        valid = checks._is_valid_geolocation(ds_valid)
+        assert np.all(valid)
+
+        # Invalid
+        valid = checks._is_valid_geolocation(ds_invalid)
+        assert np.sum(valid) == self.n_along_track - 1
+
+    def test_check_valid_geolocation(
+        self,
+        set_is_orbit_to_true: None,
+        ds_valid: xr.Dataset,
+        ds_invalid: xr.Dataset,
+    ) -> None:
+        """Test check_valid_geolocation"""
+
+        # Valid
+        checks.check_valid_geolocation(ds_valid)
+        # No error raised
+
+        # Invalid
+        with pytest.raises(ValueError):
+            checks.check_valid_geolocation(ds_invalid)
+
+    def test_has_valid_geolocation(
+        self,
+        set_is_orbit_to_true: None,
+        ds_valid: xr.Dataset,
+        ds_invalid: xr.Dataset,
+    ) -> None:
+        """Test has_valid_geolocation"""
+
+        assert checks.has_valid_geolocation(ds_valid)
+        assert not checks.has_valid_geolocation(ds_invalid)
+
+
+class TestWobblingSwath:
+    # Detect changes of direction. Repeated indices are not considered as changes
+    #               0  1  2  3  4  5  6xx7xx8xx9  10 11 12
+    lat = np.array([0, 1, 2, 2, 2, 3, 4, 3, 4, 3, 4, 5, 6])
+    n_along_track = len(lat)
+    lon = np.arange(n_along_track)
+
+    # Add cross track dimension
+    lat = lat[np.newaxis, :]
+    lon = lon[np.newaxis, :]
+
+    # Add time dimension
+    time = np.zeros(n_along_track)
+
+    # Create dataset
+    ds = xr.Dataset()
+    ds["lat"] = (("cross_track", "along_track"), lat)
+    ds["lon"] = (("cross_track", "along_track"), lon)
+    ds["gpm_granule_id"] = np.ones(n_along_track)
+    ds["time"] = time
+
+    # Threshold must be at least 3 to remove wobbling slices
+    threshold = 3
+
+    def test_get_slices_non_wobbling_swath(self) -> None:
+        """Test get_slices_non_wobbling_swath"""
+
+        returned_slices = checks.get_slices_non_wobbling_swath(self.ds, threshold=self.threshold)
+        expected_slices = [slice(0, 6 + 1), slice(9, 12 + 1)]
+        assert returned_slices == expected_slices
+
+    def test_get_slices_wobbling_swath(self) -> None:
+        """Test get_slices_wobbling_swath"""
+
+        returned_slices = checks.get_slices_wobbling_swath(self.ds, threshold=self.threshold)
+        expected_slices = [slice(7, 9)]  # TODO: check that this is the expected behavior
```

Comment:
Yes that's fine.  
I added these two utility to investigate wobbling occurence ... but it's not used anywhere in the code ... 
I noticed it often occurs when the satellite turns from an ascending to a descending orbit ...